### PR TITLE
added jpg support for images.

### PR DESCRIPTION
**Pull request by mirzazeyrek**
For saving bandwidth and space especially with huge test cases and CI tool integrations. 

screenshot_80

p.s. removed previous pr and rebased from latest state of original repo. pr is compatible with old config files. there is no need for modification.

### CODE REVIEW

**Comment by mirzazeyrek** [comment]
Please let me know if there is anything needs to be done. If you think that's okey for merge I can create another PR with one single commit :)

https://github.com/garris/BackstopJS/pull/365/files

**Comment by mirzazeyrek** [comment]
Hi @garris,

I would be very happy if you can spare some time for this.

File: capture/genBitmaps.js [hunk 1]
```diff
@@ -224,8 +225,9 @@ function processScenario (casper, scenario, scenarioOrVariantLabel, scenarioLabe
           .replace(/\{viewportLabel\}/, makeSafe(vp.name))
           .replace(/[^a-z0-9_\-]/gi, ''); // remove anything that's not a letter or a number or dash or underscore.
 
-        if (!/\.png$/i.test(fileName)) {
-          fileName = fileName + '.png';
+        var extRegExp = new RegExp(outputFormat + '$', 'i');
```
Comment (garris) [inline capture/genBitmaps.js]: maybe this is simpler/more readable?
`filename = filename.replace(outputFormat, '') + outputFormat;`

File: capture/genBitmaps.js [hunk 2]
```diff
@@ -15,6 +15,7 @@ if (!config.paths) {
   config.paths = {};
 }
 
+var outputFormat = (config.outputFormat && /^jpg$|^jpeg$/i.test(config.outputFormat)) ? '.' + config.outputFormat : '.png';
```
Comment (garris) [inline capture/genBitmaps.js]: I think this is a little cleaner?
```
var outputFormat = config.outputFormat && config.outputFormat.match(/\.?jpg$|\.?jpeg$/) || '.png';
```

File: capture/genBitmaps.js [hunk 3]
```diff
@@ -224,8 +225,9 @@ function processScenario (casper, scenario, scenarioOrVariantLabel, scenarioLabe
           .replace(/\{viewportLabel\}/, makeSafe(vp.name))
           .replace(/[^a-z0-9_\-]/gi, ''); // remove anything that's not a letter or a number or dash or underscore.
```
Comment (garris) [inline capture/genBitmaps.js]: maybe this is simpler/more readable?
`filename = filename.replace(outputFormat, '') + outputFormat;`

File: capture/genBitmaps.js (hunk 1 above)
Comment (garris) [inline capture/genBitmaps.js]: maybe this one liner is simpler/more readable?
`filename = filename.replace(outputFormat, '') + outputFormat;`

File: capture/genBitmaps.js (hunk 2 above)
Comment (mirzazeyrek) [inline capture/genBitmaps.js]: makes sense 👍

**Comment by mirzazeyrek** [comment]
Default output quality was 50 now increased it to 85. it should help with noise. I'm sending it to beta branch. Thank you for review.

**Pull request closed without merging**